File: pat1055/main.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <string>
#include <map>
using namespace std;

struct data{
    string name;
    int money;
    int age;
};

int cmp(const data& list1,const data& list2){
    if (list1.money!=list2.money){
        return list1.money > list2.money;
    }
    if (list1.age!=list2.age){
        return list1.age < list2.age;
    }
    return list1.name<list2.name;

}

int main(int argc, char** argv) {
    int n,m;
    scanf("%d%d",&n,&m);
    vector<data> saver;
    for(int i=0;i<n;i++){
        string name;
        int age, money;
        data man;
        cin>>name;
        scanf("%d%d",&age,&money);
        man.name=name;
        man.money = money;
        man.age = age;
        saver.push_back(man);
    }
    for(int i = 0;i<m;i++){
        int line,start,end;
        scanf("%d%d%d",&line,&start,&end);
        vector<data> datas;
        for(int j = 0;j<n;j++){
            if(saver[j].age>=start && saver[j].age <=end){
                datas.push_back(saver[j]);
            }
        }
        cout<<"Case #"<<i+1<<":"<<endl;
        if(datas.empty()){
            cout<<"None"<<endl;
        } else{
            sort(datas.begin(),datas.end(),cmp);
            for(int i = 0; i<line&i<datas.size();i++){
                cout<<datas[i].name<<" "<<datas[i].age<<" "<<datas[i].money<<endl;
            }
        }
    }
	return 0;
}

File: pat1055/main.py
from functools import cmp_to_key
n, m = list(map(int, input().split(' ')))
saver = []
for i in range(n):
    name, age, money = input().split(' ')
    saver.append([name, int(age), int(money)])
key_list = []
start2line = {}
def cmp(list1,list2):
    if list1[2]>list2[2]:
        return 1
    if list1[2]<list2[2]:
        return -1
    if list1[1]>list2[1]:
        return -1
    if list1[1]<list2[1]:
        return 1
    if list1[0]>list2[0]:
        return -1
    else:
        return 1
for i in range(m):
    line, start, end = list(map(int,input().split(' ')))
    datas = []
    for data in saver:
        if end>=data[1]>=start:
            datas.append(data)
    print('Case #{}:'.format(i+1))
    if len(datas) == 0:
        print('None')
    else:
        datas = sorted(datas,key=cmp_to_key(cmp),reverse=True)
        for data in datas[:line]:
            print(' '.join(map(str,data)))